Lib/glyphsLib/classes.py

```diff
@@ -3727,3 +3807,65 @@ def removeKerningForPair(self, fontMasterId, leftKey, rightKey, direction=LTR):
             del self._kerning[fontMasterId][leftKey]
         if not self._kerning[fontMasterId]:
             del self._kerning[fontMasterId]
+
+
+Number = Union[int, float]
+
+
+class LayerPointPen(AbstractPointPen):
+    """A point pen.
+
+    See :mod:`fontTools.pens.basePen` and :mod:`fontTools.pens.pointPen` for an
+    introduction to pens.
+    """
+
+    __slots__ = ("_layer", "_path")
+
+    def __init__(self, layer: GSLayer) -> None:
+        self._layer: GSLayer = layer
+        self._path: Optional[GSPath] = None
+
+    def beginPath(self, **kwargs: Any) -> None:
+        self._path = GSPath()
+
+    def endPath(self) -> None:
+        if self._path is None:
+            raise ValueError("Call beginPath first.")
+
+        self._path.closed = True
```

Comment:
endPath method does not a close a path. In pointPens, the open paths are marked by the first point of a contour having the "move" type.